packages/app/cypress/e2e/cypress-in-cypress-component.cy.ts

```diff
@@ -156,4 +164,33 @@ describe('Cypress In Cypress CT', { viewportWidth: 1500, defaultCommandTimeout:
     cy.get('#unified-runner').should('have.css', 'width', '333px')
     cy.get('#unified-runner').should('have.css', 'height', '333px')
   })
+
+  it('restarts dev server on config change', () => {
+    cy.openProject('cypress-in-cypress')
+    cy.startAppServer('component')
+    cy.visitApp()
+
+    cy.withCtx(async (ctx, { testState, sinon }) => {
+      sinon.stub(ctx._apis.projectApi.getDevServer(), 'close')
+      const devServerReady =
+        new Promise((res) => {
+          ctx._apis.projectApi.getDevServer().emitter.on('dev-server:compile:success', (err) => res(true))
```

Comment:
Are we doing anything with the value that's resolved? Should this be:

```suggestion
          ctx._apis.projectApi.getDevServer().emitter.on('dev-server:compile:success', (err) => res())
```

Reply:
Addressed in [8cabb60](https://github.com/cypress-io/cypress/pull/21212/commits/8cabb60ac598975004fbdf9e276fd6da4d78287f)